### ISSUE TITLE
Exception in starting Kernel

### ISSUE THREAD

**Issue by blackmo18** [Kotlin/kotlin-jupyter#52]
OS:
Linux Mint 19
JVM "1.8.0_201"

Error when starting jupyter cosole --kernel=kotlin

```
$ jupyter console --kernel=kotlin
[main] INFO ikotlin - Kernel args: -classpath=/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/jupyter-lib-0.7.40.145.jar:/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-1.3.70-dev-2104.jar:/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-common-1.3.70-dev-2104.jar:/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/annotations-13.0.jar, /home/blackmo/.local/share/jupyter/runtime/kernel-14107.json, -home=/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel
[main] INFO ikotlin - Starting server with config: KernelConfig(ports=[60895, 53195, 48119, 34655, 33753], transport=tcp, signatureScheme=hmac-sha256, signatureKey=d3912a08-c308643b6708e79d580871d7, pollingIntervalMillis=100, scriptClasspath=[/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/jupyter-lib-0.7.40.145.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-1.3.70-dev-2104.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-common-1.3.70-dev-2104.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/annotations-13.0.jar], resolverConfig=ResolverConfig(repositories=[kotlin.script.experimental.dependencies.RepositoryCoordinates@2aae9190, kotlin.script.experimental.dependencies.RepositoryCoordinates@2f333739, kotlin.script.experimental.dependencies.RepositoryCoordinates@77468bd9], libraries=DeferredCoroutine{Active}@71bc1ae4))
[DefaultDispatcher-worker-1] INFO ikotlin - Reading commit info for which library descriptors were cached: '/home/blackmo/.jupyter_kotlin/cache/libsCommit'
[DefaultDispatcher-worker-1] INFO ikotlin - Local libraries are cached for commit 'aebe1332e8a838318ffba9a7a06a2c565219f1d7' at '2020-01-05T19:18:11Z'
[DefaultDispatcher-worker-1] INFO ikotlin - Checking for new commits to library descriptors at https://api.github.com/repos/kotlin/kotlin-jupyter/commits?path=libraries&sha=master&since=2020-01-05T19:18:11Z
[main] DEBUG ikotlin - [hb] listen: tcp://*:60895
[main] DEBUG ikotlin - [shell] listen: tcp://*:53195
[main] DEBUG ikotlin - [control] listen: tcp://*:48119
[main] DEBUG ikotlin - [stdin] listen: tcp://*:34655
[main] DEBUG ikotlin - [iopub] listen: tcp://*:33753
[main] INFO ikotlin - Current classpath: /usr/lib/jvm/java-8-oracle/jre/lib/ext/sunec.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/sunpkcs11.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/jfxrt.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/dnsns.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/localedata.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/jaccess.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/zipfs.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/cldrdata.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/sunjce_provider.jar, /usr/lib/jvm/java-8-oracle/jre/lib/ext/nashorn.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-jupyter-kernel-0.7.40.145.jar
[main] INFO ikotlin - Begin listening for events
[main] INFO ikotlin - Classpath for compiler options: none
[DefaultDispatcher-worker-1] INFO ikotlin - No new commits to library descriptors were detected
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'spark' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/spark.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'numpy' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/numpy.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'lets-plot' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/lets-plot.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'mysql' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/mysql.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'fuel' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/fuel.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'krangl' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/krangl.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'kravis' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/kravis.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'gral' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/gral.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'kmath' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/kmath.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'klaxon' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/klaxon.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'exposed' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/exposed.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'kotlin-statistics' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/kotlin-statistics.json'
[DefaultDispatcher-worker-1] INFO ikotlin - Loading 'koma' descriptor from '/home/blackmo/.jupyter_kotlin/cache/libraries/koma.json'
[main] INFO ikotlin - Starting kotlin REPL engine. Compiler version: 1.3.70-dev-2104
[main] INFO ikotlin - Classpath used in script: [/home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/jupyter-lib-0.7.40.145.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-1.3.70-dev-2104.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/kotlin-stdlib-common-1.3.70-dev-2104.jar, /home/blackmo/.local/lib/python3.6/site-packages/run_kotlin_kernel/jars/annotations-13.0.jar]
[main] DEBUG ikotlin - [shell] >rcv: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"version":"5.3","date":"2020-03-31T00:19:51.584990Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_0"} parentHeader = {} metadata = {} content = {}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"1123adb0-9692-4920-aa7b-efa2b0904b55","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.584990Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_0"} metadata = {} content = {"execution_state":"busy"}
[main] DEBUG ikotlin - [shell] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"kernel_info_reply","msg_id":"c966a767-f0f9-4e8f-a1d6-48abc519ac88","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.584990Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_0"} metadata = {} content = {"protocol_version":"5.3","language_info":{"name":"kotlin","codemirror_mode":"text/x-kotlin","file_extension":"kt"},"implementation":"Kotlin","banner":"Kotlin language, version 1.3.70-dev-2104","language":"Kotlin","language_version":"1.3.70-dev-2104","implementation_version":"1.3.70-dev-2104","status":"ok"}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"3eb15d39-ee9f-4b5e-b6d1-309aed0b074a","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.584990Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_0"} metadata = {} content = {"execution_state":"idle"}
[main] DEBUG ikotlin - [shell] >rcv: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"version":"5.3","date":"2020-03-31T00:19:51.591735Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_1"} parentHeader = {} metadata = {} content = {}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"20287cf5-cf12-4708-a1d7-1ee80e1c97b2","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.591735Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_1"} metadata = {} content = {"execution_state":"busy"}
[main] DEBUG ikotlin - [shell] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"kernel_info_reply","msg_id":"6abcb258-a093-4c11-b174-c233fa7e2d06","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.591735Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_1"} metadata = {} content = {"protocol_version":"5.3","language_info":{"name":"kotlin","codemirror_mode":"text/x-kotlin","file_extension":"kt"},"implementation":"Kotlin","banner":"Kotlin language, version 1.3.70-dev-2104","language":"Kotlin","language_version":"1.3.70-dev-2104","implementation_version":"1.3.70-dev-2104","status":"ok"}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"e9b4fa91-5eb1-4df8-8ac4-9999cad15be7","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:51.591735Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"kernel_info_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_1"} metadata = {} content = {"execution_state":"idle"}
[main] DEBUG ikotlin - [shell] >rcv: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"version":"5.3","date":"2020-03-31T00:19:54.596428Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"history_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_2"} parentHeader = {} metadata = {} content = {"raw":true,"output":false,"hist_access_type":"tail","n":1000}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"e5f2f54a-2c67-4a62-b0f1-195d11393c48","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:54.596428Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"history_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_2"} metadata = {} content = {"execution_state":"busy"}
[main] DEBUG ikotlin - [shell] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"history_reply","msg_id":"ba6a1f52-a3d5-4b32-b456-8f0bc1b41a81","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:54.596428Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"history_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_2"} metadata = {} content = {"history":[]}
[main] DEBUG ikotlin - [iopub] snd>: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"date":"2020-03-31T08:19+0800","msg_type":"status","msg_id":"10add227-dc18-468e-88c8-5104bd652e1c","version":"5.3","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo"} parentHeader = {"version":"5.3","date":"2020-03-31T00:19:54.596428Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"history_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_2"} metadata = {} content = {"execution_state":"idle"}
Jupyter console 6.1.0

Kotlin language, version 1.3.70-dev-2104
In [1]: val hello = "hello"[main] DEBUG ikotlin - [shell] >rcv: msg[f0255f1a-e6f2ef75974643e9ac98f284] header = {"version":"5.3","date":"2020-03-31T00:20:03.574618Z","session":"f0255f1a-e6f2ef75974643e9ac98f284","username":"blackmo","msg_type":"is_complete_request","msg_id":"f0255f1a-e6f2ef75974643e9ac98f284_3"} parentHeader = {} metadata = {} content = {"code":"val hello = \"hello\""}
Exception in thread "main" java.lang.NoSuchMethodError: org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.getProject()Lorg/jetbrains/kotlin/mainKts/relocatedDeps/com/intellij/openapi/project/Project;
	at kotlin.script.experimental.jvmhost.repl.JvmReplCompiler.check(legacyReplCompilation.kt:39)
	at org.jetbrains.kotlin.jupyter.ReplForJupyter.checkComplete(repl.kt:164)
	at org.jetbrains.kotlin.jupyter.ProtocolKt.shellMessagesHandler(protocol.kt:152)
	at org.jetbrains.kotlin.jupyter.IkotlinKt.kernelServer(ikotlin.kt:104)
	at org.jetbrains.kotlin.jupyter.IkotlinKt.main(ikotlin.kt:59)
Exception in thread "Thread-3" org.zeromq.ZMQException: Errno 156384765 : Context was terminated
	at org.zeromq.ZMQ$Socket.mayRaise(ZMQ.java:1456)
	at org.zeromq.ZMQ$Socket.recv(ZMQ.java:1365)
	at org.jetbrains.kotlin.jupyter.IkotlinKt$kernelServer$$inlined$use$lambda$1.invoke(ikotlin.kt:123)
	at org.jetbrains.kotlin.jupyter.IkotlinKt$kernelServer$$inlined$use$lambda$1.invoke(ikotlin.kt)
	at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
In [1]: val hello = "hello"                                                                                                                                                                                                                   
/home/blackmo/.local/lib/python3.6/site-packages/jupyter_console/ptshell.py:656: UserWarning: The kernel did not respond to an is_complete_request. Setting `use_kernel_is_complete` to False.
  warn('The kernel did not respond to an is_complete_request.
```

**Comment by ileasile** [comment]
Hello! Thank you for reporting this. The error was fixed in build **0.7.41.254.dev1** which may be installed via
`conda install -c jetbrains-dev kotlin-jupyter-kernel==0.7.41.254.dev1`
By the way, debug logging is still annoying. I hope, we will hide logging for console mode or let the user to setup logging level by magic or by editing some file.

**Comment by ileasile** [comment]
Related issue: #54
